src/br/ufpe/cin/logging/LoggerStatistics.java

```diff
@@ -20,6 +20,8 @@
 
 public class LoggerStatistics {
 
+	public static int recursionCounter = 0;
```

Comment:
O código dá a entender que em caso de exceção, ele vai permitir reinicializar o processo (logContext, logSummary, etc) no máximo 'recursionCounter' vezes? É isso? 

o 'recursionCounter' serve para então evitar uma recursão infinita. Se for isso, adiciona um comentário explicando nesse atributo.